Mathlib/Algebra/GradedMonoid.lean

```diff
@@ -582,6 +582,27 @@ variable {S : Type*} [SetLike S R] [Monoid R] [AddMonoid ι]
 
 variable {A : ι → S} [SetLike.GradedMonoid A]
 
+namespace GradeZero
+variable (A) in
+
+/-- The submonoid `A 0` of `R`. -/
+@[simps]
+def submonoid : Submonoid R where
+  carrier := A 0
+  mul_mem' ha hb := add_zero (0 : ι) ▸ SetLike.mul_mem_graded ha hb
+  one_mem' := SetLike.one_mem_graded A
+
+/-- The monoid `A 0` inherited from `R` in the presence of `SetLike.GradedMonoid A`. -/
+instance instMonoid : Monoid (A 0) := inferInstanceAs <| Monoid (GradeZero.submonoid A)
+
+@[simp, norm_cast] theorem coe_one : ↑(1 : A 0) = (1 : R) := rfl
+
+@[simp, norm_cast] theorem coe_mul (a b : A 0) : ↑(a * b) = (↑a * ↑b : R) := rfl
+
+@[simp, norm_cast] theorem coe_pow (a : A 0) (n : ℕ) : ↑(a ^ n) = (↑a : R) ^ n := rfl
```

Comment:
@FMLJohn: whenever you write a new instance, you should write lemmas like this (proven with `rfl`) that characterize it.

Note that we don't need to repeat ourselves for the semiring and ring instances, since they all use the same definition of `*`/`1`/`^` so these lemmas match all three of them.